From 1c96469d0d329bf610ce4c2660363b1e310cf70d Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:06:37 -0400
Subject: [powerpc] vio: use strcpy in modalias_show

Message-id: <1410545655-205645-169-git-send-email-dzickus@redhat.com>
Patchwork-id: 94176
O-Subject: [RHEL7 PATCH 168/626] powerpc/vio: use strcpy in modalias_show
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 411cabf79e684171669ad29a0628c400b4431e95
Author: Prarit Bhargava <prarit@redhat.com>
Date:   Thu Oct 17 08:00:11 2013 -0400

    powerpc/vio: use strcpy in modalias_show

    Commit e82b89a6f19bae73fb064d1b3dd91fcefbb478f4 used strcat instead of
    strcpy which can result in an overflow of newlines on the buffer.

    Signed-off-by: Prarit Bhargava
    Cc: benh@kernel.crashing.org
    Cc: ben@decadent.org.uk
    Cc: stable@vger.kernel.org
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/vio.c b/arch/powerpc/kernel/vio.c
index d38cc08..cb92d82 100644
--- a/arch/powerpc/kernel/vio.c
+++ b/arch/powerpc/kernel/vio.c
@@ -1531,12 +1531,12 @@ static ssize_t modalias_show(struct device *dev, struct device_attribute *attr,
 
  dn = dev->of_node;
  if (!dn) {
-  strcat(buf, "\n");
+  strcpy(buf, "\n");
   return strlen(buf);
  }
  cp = of_get_property(dn, "compatible", NULL);
  if (!cp) {
-  strcat(buf, "\n");
+  strcpy(buf, "\n");
   return strlen(buf);
  }
 
-- 
1.7.1